pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableBuilder.java

```diff
@@ -77,6 +77,9 @@
 // TODO:   3. Given a data schema, write all values one by one instead of using rowId and colId to position (save time).
 // TODO:   4. Store bytes as variable size data instead of String
 public class DataTableBuilder {
```

Comment:
Suggest making 2 builders, one for v2 and one for v3. You can extract the common logic into a base class, or just duplicate code because we will deprecate v2 in the next release once v3 is well tested

Reply:
We plan to remove all V2 logic after the next release. So, we can keep re-factors and beautifications to a minimum. Please do only what is necessary because all of V2 logic will disappear and someone looking at the code will wonder why we have a base class

Reply:
+1 for keeping the current logic. Another drawback of having two builder is: all caller need to decide call v2 builder or v3 builder based on instance config, which is ugly.

Reply:
+1 for keeping current logic. Another drawback of have two builder is: all caller need to decide whether to call V2/V3 based on instance config, which is ugly.